bin/amphtml-update.py

```diff
@@ -211,6 +211,23 @@ def GenerateFooterPHP(out):
 		return self::$allowed_tags;
 	}
 
+	/**
+	 * Get extension specs.
+	 *
+	 * @since 1.5
+	 * @internal
+	 * @return array Extension specs, keyed by extension name.
+	 */
+	public static function get_extension_specs() {
+		$extension_specs = [];
+		foreach ( self::get_allowed_tag( 'script' ) as $script_spec ) {
+			if ( isset( $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec'] ) ) {
+				$extension_specs[ $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['name'] ] = $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec'];
+			}
+		}
+		return $extension_specs;
+	}
```

Comment:
No need to process this every time it is requested (which is at least _twice_ per request).
```php
public static function get_extension_specs() {
	static $extension_specs = [];

	if ( ! empty( $extension_specs ) ) {
		return $extension_specs;
	}

	foreach ( self::get_allowed_tag( 'script' ) as $script_spec ) {
		if ( isset( $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec'] ) ) {
			$extension_specs[ $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['name'] ] = $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec'];
		}
	}

	return $extension_specs;
}
```

Reply:
Good point. Done in 16f35b4.